### ISSUE TITLE
Question regarding the generation of adversarial examples

### ISSUE THREAD

**Issue by Sammy42779** [yunqing-me/AttackVLM#9]
Hello! First and foremost, I'd like to extend my gratitude for sharing your outstanding work.

I have a question regarding the generation of adversarial examples through transfer-based attacks. 
Within the comments in the code `_train_adv_img_blip.py`, it is stated that the "image is normalized to (0.0, 1.0)". However, upon debugging, I observed that the values were indeed normalized.

<img width="950" alt="image" src="https://github.com/yunqing-me/AttackVLM/assets/12605256/47a43359-dab4-42ef-8a72-7d45001eb661">

Moreover, I noticed the presence of a scaling_tensor as illustrated below:
```
    # for normalized imgs
    scaling_tensor = torch.tensor((0.26862954, 0.26130258, 0.27577711), device=device)
    scaling_tensor = scaling_tensor.reshape((3, 1, 1)).unsqueeze(0)
    alpha = args.alpha / 255.0 / scaling_tensor
    epsilon = args.epsilon / 255.0 / scaling_tensor
```

Is it correct to infer that the `scaling_tensor` is used to normalize both `alpha` and `epsilon`? If so, does this mean that we can directly employ normalized data without resorting to the previous method that requires constraining the image values within the [0,1] range? Would you kindly clarify this aspect?

Thank you in anticipation of your timely response.

**Comment by yunqing-me** [comment]
Thank you for your interest in our work. Regarding your question:

1): images are converted to [0, 1] via torchvision ToTensor() functions before normalization (which is a standard transformation)

2): the `scaling_tensor` is used accordingly to normalize other parameters when you are tackling normalized images
